### ISSUE TITLE
Private members

### ISSUE THREAD

**Issue by qwabra** [microsoft/TypeScript#41773]
**TypeScript Version:**  4.1.2

**Search Terms:**

**Expected behavior:**

should throw an error with `"target": "ESNext"` and `#` private member

**Actual behavior:**

**Related Issues:**

https://github.com/Microsoft/TypeScript/issues/17535
https://stackoverflow.com/a/45244695/4794368

**Code**
```ts
class A {
    #str = 'pr Str'
    constructor() {
        new A.B(this)
    }
    private static B = class {
        constructor(private a: A) {
            console.log(this.a.#str)
        }
    }
}
test: {
    new A
}

```

<details><summary><b>Output</b></summary>

```ts
"use strict";
class A {
    constructor() {
        this.#str = 'pr Str';
        new A.B(this);
    }
    #str;
}
A.B = class {
    constructor(a) {
        this.a = a;
        console.log(this.a.#str);
    }
};
test: {
    new A;
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "alwaysStrict": true,
    "allowUnusedLabels": true,
    "esModuleInterop": true,
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": 2,
    "target": "Latest",
    "jsx": "React",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play?allowUnusedLabels=true&target=99&ssl=1&ssc=1&pln=15&pc=1#code/MYGwhgzhAECC0G8BQ1XQMQQC4CdoF5oByABzwGVciU1gB7AO2xwFdgs6cAKASkRrRoGAUwDucAHQAhLlgAWASwg8B0AL6qyCgG5gsw6Nj0Lg0KQWihIMZIMH0muNh25bd+6GABccPrbsBDhB0IMISIHQA5rKKEBJgEpi4KgGoGoIaGvrYPv6oIuKwSBpAA)

**Comment by RyanCavanaugh** [comment]
Why? This code runs without error
```js
function f() {
    "use strict";
    class A {
        #str = 'pr Str'
        constructor() {
            new A.B(this)
        }
        static B = class {
            constructor(a) {
                console.log(a.#str)
            }
        }
    }

    new A();
}
```
![image](https://user-images.githubusercontent.com/6685088/100904476-e99e0080-347b-11eb-9c28-0f9a76d44691.png)

**Comment by qwabra** [comment]
Oh, hmm, hmmm but.

But your example is different from the compiler output

**Comment by RyanCavanaugh** [comment]
Not really seeing any argument being made here. Private identifiers have lexical visibility and `#str` is in `B`'s containing lexical scope.

**Comment by qwabra** [comment]
ok, now i see. but the generated code doesn't work. please look at the output

**Comment by DanielRosenwasser** [comment]
Thanks for clarifying - I opened up https://github.com/microsoft/TypeScript/issues/41788 for you.
